src/refract/JSONSchemaVisitor.cc

```diff
@@ -340,62 +336,22 @@ namespace refract
 
     void JSONSchemaVisitor::visit(const NullElement& e)
     {
-        NullElement nil;
-        addMember("type", &nil);
+        addMember("type", new NullElement);
```

Comment:
Original code was passing a pointer to a local variable allocated on stack ... and yes there is clean up, when the Visitor's desctructor is called.